### PR TITLE
Add documents for Windows

### PR DESCRIPTION
**Pull request by peterjc123**

### CODE REVIEW

File: docs/source/notes/windows.rst [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+Include optional components
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There're two supported components for Windows PyTorch:
+MKL and MAGMA. Here are the steps to build with them.
+
+.. code-block:: bat
+
+    REM Make sure you has 7z and curl installed.
```
Comment (nzw0301) [inline docs/source/notes/windows.rst]: `has` → `have`

File: docs/source/notes/windows.rst [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    set "CMAKE_INCLUDE_PATH=%cd%\\mkl\\include"
+    set "LIB=%cd%\\mkl\\lib;%LIB%"
+    set "MAGMA_HOME=%cd%\\magma"
+
+Speeding CUDA build for Windows
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Visual Studio doesn't support parallel custom task currently.
+As an alternative, we can use ``Ninja`` to parallelize CUDA
+build tasks. It can used by typing only a few lines of code.
```
Comment (nzw0301) [inline docs/source/notes/windows.rst]: `can be used`

File: docs/source/notes/windows.rst [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    - https://repo.continuum.io/pkgs/msys2/win-32
+    - https://repo.continuum.io/pkgs/msys2/noarch
+
+PyTorch doesn't work on 32-bit system. Please use Windows and
+Python 64-bit version.
+
+Why are there no Python 2 packages for Windows?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Because it's not stable enough. There're some issues that needs to
```
Comment (nzw0301) [inline docs/source/notes/windows.rst]: `need`

File: docs/source/notes/windows.rst [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+.. code-block:: py3tb
+
+   THCudaCheck FAIL file=torch\csrc\generic\StorageSharing.cpp line=252 error=63 : OS call failed or operation not supported on this OS
+
+They are not supported on Windows. Something like doing multiprocessing on CUDA
+tensors cannot succeed, there are two alternatives for this.
+
+1. Don't use `multiprocessing`. Set the ``num_worker`` of ``DataLoader`` to zero.
+2. Share CPU tensors instead. Remove ``.cuda()`` in your custom DataLoader definetion.
```
Comment (nzw0301) [inline docs/source/notes/windows.rst]: `definetion` → `definition`

File: docs/source/notes/windows.rst [hunk 5]
```diff
@@ -0,0 +1,256 @@
+Windows Usage
```
Comment (apaszke) [inline docs/source/notes/windows.rst]: Let's call it Windows FAQ

File: docs/source/notes/windows.rst [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+    REM Set it as the cmake generator
+    set CMAKE_GENERATOR=Ninja
+
+
+One key install script
+^^^^^^^^^^^^^^^^^^^^^^
+
+You can take a look at the script `here
+<https://github.com/peterjc123/pytorch-scripts>`_. 
+It will lead the way for you.
```
Comment (apaszke) [inline docs/source/notes/windows.rst]: Is building from source really as compilcated as having a bunch of build scripts? What do they do, and why isn't is as simple as `python setup.py install`, like on Linux/macOS?

File: docs/source/notes/windows.rst [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+
+.. code-block:: py3tb
+
+   THCudaCheck FAIL file=torch\csrc\generic\StorageSharing.cpp line=252 error=63 : OS call failed or operation not supported on this OS
+
+They are not supported on Windows. Something like doing multiprocessing on CUDA
+tensors cannot succeed, there are two alternatives for this.
+
+1. Don't use `multiprocessing`. Set the ``num_worker`` of ``DataLoader`` to zero.
+2. Share CPU tensors instead. Remove ``.cuda()`` in your custom DataLoader definition.
```
Comment (SsnL) [inline docs/source/notes/windows.rst]: nit1: It should be custom `Dataset` definition (or better ``:class:`~torch.utils.data.Dataset` ``).

nit: Can you also mention the similar effect `.to()` calls? Or maybe we should just say that the dataset should return CPU tensors.

File: docs/source/notes/windows.rst [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: py3tb
+
+   THCudaCheck FAIL file=torch\csrc\generic\StorageSharing.cpp line=252 error=63 : OS call failed or operation not supported on this OS
+
+They are not supported on Windows. Something like doing multiprocessing on CUDA
+tensors cannot succeed, there are two alternatives for this.
+
+1. Don't use `multiprocessing`. Set the ``num_worker`` of ``DataLoader`` to zero.
```
Comment (SsnL) [inline docs/source/notes/windows.rst]: nit: double tick for ``` ``multiprocessing`` ``` for it to be rendered like code.

File: docs/source/notes/windows.rst [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+uses ``spawn`` instead of ``fork``. So we have to wrap the code with an
+if-clause to protect the code from executing multiple times. Refactor
+your code into the following structure.
+
+.. code-block:: python
+
+    import torch
+
+    def main()
+        for i, (x, y) in dataloader:
```
Comment (SsnL) [inline docs/source/notes/windows.rst]: nit: you need `enumerate` to get `i`. Also can we just say `data` instead of `(x, y)`

**Review by ezyang — APPROVED** [review]

**Pull request merged**